server.go

```diff
@@ -96,21 +187,21 @@ var upgrader = websocketConn.Upgrader{
 }
 
 func (s *Server) listen() {
-	lis, err := net.Listen("tcp", fmt.Sprintf(":%v", s.grpcPort))
+	lis, err := net.Listen("tcp", fmt.Sprintf(":%v", s.GrpcPort))
 	if err != nil {
 		log.Fatalf("failed to listen: %v", err)
 	}
-	grpcServer := grpc.NewServer()
-	proto.RegisterSlaveRegistrationServiceServer(grpcServer, s)
+	s.grpcServer = grpc.NewServer()
+	proto.RegisterSlaveRegistrationServiceServer(s.grpcServer, s)
 
 	go func() {
-		if err := grpcServer.Serve(lis); err != nil {
+		if err := s.grpcServer.Serve(lis); err != nil {
 			log.Fatalf("failed to serve: %v", err)
 		}
 	}()
 
 	http.HandleFunc("/", s.websocketHandler)
-	log.Fatal(http.ListenAndServe(fmt.Sprintf(":%v", s.httpPort), nil))
+	log.Fatal(http.ListenAndServe(fmt.Sprintf(":%v", s.HttpPort), nil))
```

Comment:
Bonus points: You can use a `http.Server` here that we can explicitly shutdown: 
https://stackoverflow.com/questions/39320025/how-to-stop-http-listenandserve

Reply:
the func http.Server.Shutdown needs a context as a parameter, which context should we use?

Reply:
https://golang.org/pkg/net/http/#Server.Shutdown
you can use a context from `s.withTimeout` (timeout can be something like 1*time.Second)
We should probably close the connected websocket connections too, so we need a shtdown hook in https://github.com/codeuniversity/al-master/blob/master/websocket/connections_handler.go

Reply:
So I have multiple problems: 
**1)** I don't see how I can pass the http.Server.Shutdown() function to your withTimeout() method because it has "error" as return type but that 's not a problem I can just create the context manually with `context.WithTimeout(context.Background(), 1*time.Second)` 
 **2)** Everytime I call s.httpServer.Shutdown() I get an error which says "http: Server closed" so for some reason the server is already closed before I try shut it down 
**3)** I'm not sure what you mean with "shutdown hook". 
-> I think I'll just postpone that task until we're on campus again

Reply:
1) that works too (just make sure to call cancel at some point)
2) are you sure the server is already running?
3) I mean a Shutdown func defined on the connnections handler, that closes all connected websockets connections
We can postpone this until later :)

Reply:
1. but I get `Cannot use 's.httpServer.Shutdown' (type func(ctx context.Context) error) as type func(ctx context.Context)` if I try to pass
2. solved it, the problem was the `log.Fatal()` which made the program exit immediately when the server got closed. Is it a problem if I change it to `log.Println()` instead?
3. ahhh so a new function `func (c *ConnectionsHandler)` in connections_handler.go which we call from server.go where we loop through the connections and close them?

Reply:
1) I'm not sure what you mean
2) that is fine
3) yes (and also stop the listen goroutine in each connection)